src/aura/AuraAvatarTwoToken.sol

```diff
@@ -587,47 +595,79 @@ contract AuraAvatarTwoToken is BaseAvatar, PausableUpgradeable, AuraAvatarUtils,
+        }
+
+        _processRewards(_auraPriceInUsd);
+    }
+
     /// @notice Claim and process BAL and AURA rewards, selling some of it to USDC and depositing the rest to bauraBAL
     ///         and vlAURA.
     /// @param _auraPriceInUsd A reference price for AURA in USD (in 8 decimal precision) to compare with TWAP price. Leave as 0 to use only TWAP.
     /// @return processed_ An array containing addresses and amounts of harvested tokens (i.e. tokens that have finally
     ///                    been swapped into).
-    function processRewardsInternal(uint256 _auraPriceInUsd) internal returns (TokenAmount[] memory processed_) {
+    function _processRewards(uint256 _auraPriceInUsd) internal returns (TokenAmount[] memory processed_) {
         // 1. Claim BAL and AURA rewards
         (uint256 totalBal, uint256 totalAura) = claimAndRegisterRewards();
 
         // 2. Swap some for USDC and send to owner
         uint256 balForUsdc = (totalBal * sellBpsBalToUsdc) / MAX_BPS;
         uint256 auraForUsdc = (totalAura * sellBpsAuraToUsdc) / MAX_BPS;
 
-        uint256 usdcEarnedFromBal = swapBalForUsdc(balForUsdc);
-        uint256 usdcEarnedFromAura = swapAuraForUsdc(auraForUsdc, _auraPriceInUsd);
+        uint256 usdcEarnedFromBal;
+        if (balForUsdc > 0) {
+            usdcEarnedFromBal = swapBalForUsdc(balForUsdc);
+        }
+        uint256 usdcEarnedFromAura;
+        if (auraForUsdc > 0) {
+            usdcEarnedFromAura = swapAuraForUsdc(auraForUsdc, _auraPriceInUsd);
+        }
 
         uint256 totalUsdcEarned = usdcEarnedFromBal + usdcEarnedFromAura;
 
         address ownerCached = owner();
-        USDC.safeTransfer(ownerCached, totalUsdcEarned);
+        if (totalUsdcEarned > 0) {
+            USDC.safeTransfer(ownerCached, totalUsdcEarned);
+        }
 
         // 3. Deposit remaining BAL to 80BAL-20ETH BPT
-        depositBalToBpt(totalBal - balForUsdc);
+        uint256 balToBpt = totalBal - balForUsdc;
+        if (balToBpt > 0) {
+            depositBalToBpt(totalBal - balForUsdc);
+        }
 
         // 4. Swap BPT for auraBAL or lock
-        swapBptForAuraBal(BPT_80BAL_20WETH.balanceOf(address(this)));
+        uint256 bptToAuraBal = BPT_80BAL_20WETH.balanceOf(address(this));
+        if (bptToAuraBal > 0) {
+            swapBptForAuraBal(bptToAuraBal);
+        }
 
         // 5. Dogfood auraBAL in Badger vault on behalf of owner
-        uint256 bauraBalBefore = BAURABAL.balanceOf(ownerCached);
-        BAURABAL.depositFor(ownerCached, AURABAL.balanceOf(address(this)));
-        uint256 bauraBalAfter = BAURABAL.balanceOf(ownerCached);
+        uint256 auraBalBalance = AURABAL.balanceOf(address(this));
+        uint256 bAuraBalEarned;
+        if (auraBalBalance > 0) {
+            uint256 bAuraBalBefore = BAURABAL.balanceOf(ownerCached);
+            BAURABAL.depositFor(ownerCached, auraBalBalance);
+            bAuraBalEarned = BAURABAL.balanceOf(ownerCached) - bAuraBalBefore;
+        }
```

Comment:
Same with this block, wouldn't it make sense to merge steps 3, 4 and 5 into the same block and use only one balance check? In the end, `balToBpt` will be more than 0 if, only if there was any BAL to begin with and we didn't sell it all for USDC.